### PR TITLE
Fix 254 delete last elem list page

### PR DESCRIPTION
**Pull request by kialam**
##### SUMMARY

Related #4239 

This PR does the following:

Updates the itemCount of the Orgs and Templates list page when an org/template has been successfully deleted.
Add a new method `getCurrPage()` to `PaginatedDataList` that updates pagination and query params accordingly when the current page a user is on is greater than the total pages the component is supposed to display.
Add unit test for `getCurrPage()` functionality.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - UI

### CODE REVIEW

**Comment by mabashian** [comment]
Link https://github.com/ansible/awx/issues/4239

**Comment by jakemcdermott** [comment]
recheck

File: awx/ui_next/src/components/PaginatedDataList/PaginatedDataList.jsx [hunk 1]
```diff
@@ -47,6 +55,21 @@ class PaginatedDataList extends React.Component {
...
+    const { qsConfig, location } = this.props;
+    const { page_size, page: currPage } = parseNamespacedQueryString(
+      qsConfig,
+      location.search
+    );
+    const lastPage = Math.ceil(itemCount / page_size);
+    if (currPage > lastPage) {
+      this.pushHistoryState({ page: lastPage || 1 });
+    }
+  }
```
Comment (jlmitch5) [inline awx/ui_next/src/components/PaginatedDataList/PaginatedDataList.jsx:71]: nice

**Review by jlmitch5 — APPROVED** [review]
looks good!

**Pull request merged**